File: lua/CSharp/LuaParserBase.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using Antlr4.Runtime;

public abstract class LuaParserBase : Parser
{
    const bool debug = false;

    protected LuaParserBase(ITokenStream input)
        : base(input)
    {
    }

    protected LuaParserBase(ITokenStream input, TextWriter output, TextWriter errorOutput)
        : base(input, output, errorOutput)
    {
    }

    protected void myreset()
    {
	base.Reset();
    }

    public bool IsFunctionCall()
    {
        var la = this.TokenStream.LT(1);
	if (la.Type != LuaParser.NAME) return false;
	la = this.TokenStream.LT(2);
	if (la.Type == LuaParser.OP) return false;
	return true;
    }
}

File: lua/Dart/LuaParserBase.dart
import 'package:antlr4/antlr4.dart';
import 'LuaParser.dart';
import 'dart:collection';

abstract class LuaParserBase extends Parser
{
    final bool debug = false;

    LuaParserBase(TokenStream input) : super(input);

    bool IsFunctionCall()
    {
        final la = this.tokenStream.LT(1);
        if (la?.type != LuaParser.TOKEN_NAME) return false;
        final la2 = this.tokenStream.LT(2);
        if (la2?.type == LuaParser.TOKEN_OP) return false;
	return true;
    }
}
